.github/workflows/functional-test.yml

```diff
@@ -1,5 +1,8 @@
 name: Functional Tests
-on: [workflow_dispatch, pull_request]
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: [ main ]
```

Comment:
Functional tests will run only for PR's against `main` or manually (on demand) via Github UI. Running against older releases shouldn't be necessary as Magento Marketplace must only test the latest version on publish.